Java/src/main/java/net/hypixel/api/util/PropertyFilter.java

```diff
@@ -0,0 +1,309 @@
+                }
+            }
+
+            if (shouldAddKey) {
+                allowedKeys.add(key);
+            }
+        }
+    }
+
+    /**
+     * Removes all of the provided keys from the filter, such that objects {@link
+     * #applyTo(ComplexHypixelObject) passed through} the filter will <em>not</em> include
+     * properties with those keys.
+     * <p><br>
+     * Attempting to remove a key that was already removed, or never {@link #include(String...)
+     * included} to begin with, will have no effect.
+     */
+    public void removeKeys(String... keys) {
+        if (keys == null) {
+            throw new IllegalArgumentException("Cannot remove null keys");
+        }
+
+        for (String key : keys) {
+            if (key == null) {
+                throw new IllegalArgumentException("Cannot remove null keys");
+            }
+            allowedKeys.removeIf(existingKey -> existingKey.toString().equals(key));
+        }
+    }
+
+    /**
+     * @return A new set containing all property keys that can {@link #applyTo(ComplexHypixelObject)
+     * pass through} the filter.
+     * @see #include(String...)
+     */
+    public Set<String> getIncludedKeys() {
+        Set<String> fullKeys = new HashSet<>(allowedKeys.size());
+        for (PropertyKey key : allowedKeys) {
+            fullKeys.add(key.toString());
+        }
+        return fullKeys;
+    }
+
+    /**
+     * {@link #applyTo(ComplexHypixelObject) Applies} the filter to all {@code objects} provided as
+     * arguments.
+     *
+     * @throws IllegalArgumentException If no objects are provided, or if any of the objects are
+     *                                  {@code null}.
+     */
+    public void applyTo(ComplexHypixelObject... objects) {
+        if (objects == null || objects.length == 0) {
+            throw new IllegalArgumentException("Batch filtering requires at least 1 object");
+        }
+
+        for (ComplexHypixelObject object : objects) {
+            applyTo(object);
+        }
+    }
+
+    /**
+     * {@link #applyTo(ComplexHypixelObject) Applies} the filter to any {@code objects} iterated
+     * over.
+     *
+     * @throws IllegalArgumentException If any object received during iteration is {@code null}.
+     */
+    public void applyTo(Iterable<ComplexHypixelObject> objects) {
+        for (ComplexHypixelObject object : objects) {
+            applyTo(object);
+        }
+    }
+
+    /**
+     * Strips the {@code object} of any properties that haven't explicitly been allowed via {@link
+     * #include(String...)} or the {@link PropertyFilter#with(String...) with(...) constructor}.
+     * <p><br>
+     * The resulting object will (at most) only contain the properties returned by {@link
+     * #getIncludedKeys()}. Any properties missing from the object will not be added.
+     *
+     * @throws IllegalArgumentException If the {@code object} is {@code null}.
+     */
+    public void applyTo(ComplexHypixelObject object) {
```

Comment:
I believe this is completely backwards. A PropertyFilter should have no idea what a ComplexHypixelObject is. A ComplexHypixelObject should provide a way to apply a given filter to itself, not the other way around where a filter applies itself to an object

Reply:
It might be another matter of preference then. To me, the filter analogy makes more sense when the object is what is being passed through the filter. Kinda like how sand exists just fine on its own, but if you happen to have a filter around, you might find something better inside.

Calling it "applying" was probably bad naming on my part, but I wasn't sure if `PropertyFilter#filter(...)` would be too redundant. Let me know what you think.

Reply:
Oh, I can understand what you mean there. Personally though, I see the object as an entity that gives you ways to access it and modify it, and filtering in this case looks like something an object would allow you to do to it, with the "filter rules" being given to it. I believe this pattern is more prevalent and would be easier for our clients to learn

Reply:
That's fair. I've flipped this the other way around, with `UnstableHypixelObject#filter()` replacing `PropertyFilter#applyTo()`. PropertyFilter no longer refers to Unstable, other than some doc examples using player-related keys.